### PR TITLE
Fix last models for common tests that are too big.

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR fixes the last batch of models that are too big for common tests. After it, only two classes override the common test to skip it:
- timm-backbone
- layoutlmv2
In the first case, it's because there is no timm model small enough to work (I did switch to the smallest resnet though) and in the second detectron2 does not let us configure a smaller backbone (I did try with the constant exposed by they don't seem to have any effect).

### CODE REVIEW

File: src/transformers/models/perceiver/modeling_perceiver.py [hunk 1]
```diff
@@ -1830,7 +1830,7 @@ def __init__(self, config: PerceiverConfig):
             # Autoencoding, don't pass inputs to the queries.
             concat_preprocessed_input=False,
             output_shape=config.output_shape,
-            output_num_channels=512,
+            output_num_channels=config.output_num_channels,
```
Comment (sgugger) [inline src/transformers/models/perceiver/modeling_perceiver.py:1833]: The changes here are just to enable the configuration of smaller models instead of hard-coding dims at 512 and 1024.

File: tests/models/layoutlm/test_modeling_layoutlm.py [hunk 2]
```diff
@@ -279,10 +279,6 @@ def test_for_question_answering(self):
         config_and_inputs = self.model_tester.prepare_config_and_inputs()
         self.model_tester.create_and_check_for_question_answering(*config_and_inputs)
 
-    @unittest.skip("Will be fixed soon by reducing the size of the model used for common tests.")
```
Comment (sgugger) [inline tests/models/layoutlm/test_modeling_layoutlm.py:282]: Here I skipped by mistake, it was already small enough.

File: tests/test_modeling_common.py [hunk 3]
```diff
@@ -2708,7 +2708,7 @@ def test_load_with_mismatched_shapes(self):
     def test_model_is_small(self):
         # Just a consistency check to make sure we are not running tests on 80M parameter models.
         config, _ = self.model_tester.prepare_config_and_inputs_for_common()
-        # print(config)
+        print(config)
```
Comment (ydshieh) [inline tests/test_modeling_common.py:2711]: guess you will remove it before merge :-)

File: src/transformers/models/table_transformer/configuration_table_transformer.py [hunk 4]
```diff
@@ -237,6 +237,17 @@ def num_attention_heads(self) -> int:
...
+        """
+        Serializes this instance to a Python dictionary. Override the default [`~PretrainedConfig.to_dict`]. Returns:
+            `Dict[str, any]`: Dictionary of all the attributes that make up this configuration instance,
+        """
+        output = copy.deepcopy(self.__dict__)
+        if output["backbone_config"] is not None:
+            output["backbone_config"] = self.backbone_config.to_dict()
+        output["model_type"] = self.__class__.model_type
+        return output
+
```
Comment (ydshieh) [inline src/transformers/models/table_transformer/configuration_table_transformer.py:250]: Could you remind me why this change is necessary to allow using smaller model? Thanks
(I am OK with the change, just want to know a bit the detail)

**Review by ydshieh — APPROVED** [review]
Very nice effort! Thanks as always.

File: src/transformers/models/table_transformer/configuration_table_transformer.py (hunk 4 above)
Comment (sgugger) [reply src/transformers/models/table_transformer/configuration_table_transformer.py:250]: This is something that should have been done when adding the model. You can't serialize the config otherwise (so you can't print it for instance) when it has a `backbone_config`.

File: tests/test_modeling_common.py (hunk 3 above)
Comment (sgugger) [reply tests/test_modeling_common.py:2711]: Removed!

**Review by amyeroberts — APPROVED** [review]
Thanks for working on this and improving our tests!
